platforms/ios/framework/src/TGMapView.h

```diff
@@ -399,6 +399,26 @@ TG_EXPORT
  */
 @property (strong, nonatomic) UILongPressGestureRecognizer* longPressGestureRecognizer;
 
+#pragma mark Map region change state notifiers
+
+/**
+ Client can use these to trigger explicit `TGMapRegionChangeStates` changes
+ Notifies that client is going to begin map region animation
+ */
+- (void)notifyGestureDidBegin;
```

Comment:
So, the general iOS guideline for callbacks is
`will` / `is` / `did` and you want to pass along the source, so normally the map view. For example:
`- (void)mapView:(TGMapView*)mapview willChangeRegion;`
`- (void)mapView:(TGMapView*)mapview regionChanged;`
`- (void)mapView:(TGMapView*)mapview didFinishChangingRegion;`

You might also want to think about, similar to gesture recognizers, something like:
`- (BOOL)shouldMapview:(TGMapView*)mapview allowRegionChange`

Another thought: I would say the naming here is a little confusing from an iOS perspective. Notification is generally done via delegate callbacks, NSNotificationCenter, or KVO. Function callbacks *do* happen, like in `viewDidLoad` and other lifecycle-ish methods, but generally it's preferable to go with one of the other 3.

Lastly, if you plan on keeping these, I would add comments here about subclassing requirements that developers *must* call super, or internal functionality will break if you don't.

Reply:
Discussed this in person with @tallytalwar and understand the decisions here now :)